OpenRA.Game/CPos.cs

```diff
@@ -19,23 +19,22 @@ namespace OpenRA
 	public struct CPos : IScriptBindable, ILuaAdditionBinding, ILuaSubtractionBinding, ILuaEqualityBinding, ILuaTableBinding, IEquatable<CPos>
 	{
 		public readonly int X, Y;
+		public readonly int Layer;
```

Comment:
Encoding `Layer` as a byte here means we can safely do the same elsewhere where it might actually save you something (if you had some collection of layer indexes, for.... something).

Besides, 256 layers ought to be enough for anybody.